cluster-autoscaler/core/scaledown/legacy/wrapper.go

```diff
@@ -47,7 +51,15 @@ func NewScaleDownWrapper(sd *ScaleDown) *ScaleDownWrapper {
 func (p *ScaleDownWrapper) UpdateClusterState(podDestinations, scaleDownCandidates []*apiv1.Node, actuationStatus scaledown.ActuationStatus, pdbs []*policyv1.PodDisruptionBudget, currentTime time.Time) errors.AutoscalerError {
 	p.sd.CleanUp(currentTime)
 	p.pdbs = pdbs
-	return p.sd.UpdateUnneededNodes(podDestinations, scaleDownCandidates, currentTime, pdbs)
+	err := p.sd.UpdateUnneededNodes(podDestinations, scaleDownCandidates, currentTime, pdbs)
+	if err != nil {
+		return err
+	}
+	_, drained := actuationStatus.DeletionsInProgress()
+	if len(drained) > 0 {
```

Comment:
Hmm, wouldn't it be easier to do this check in StartDeletion instead? This way you wouldn't need to check error type in static_autoscaler. With that solution you could also move `scaleDownStatus.Result = status.ScaleDownInProgress` into the wrapper (IIRC we don't think we want to retain ScaleDownInProgress status in new implementation?)

Reply:
There's also a second difference - putting this in Actuator would cause `processors.NodeGroupManager.RemoveUnneededNodeGroups` to be called when there's an ongoing drain, which would change the existing behavior of legacy scale down.

Reply:
I hoped we may be able to make static_autoscaler not aware of legacy / new scale-down and just use the interface. I'd rather have a permanent actuator wrapper for legacy scale-down or something along those lines if that meant we can clean up static_autoscaler. Ideally we'd encapsulate all logic related to legacy inside the broader scale-down package.

If we need to spill over to static_autoscaler for other reasons than I guess I'm fine with leaving this as-is.